.github/workflows/bundleTest-thor.yml

```diff
@@ -0,0 +1,238 @@
+        shell: "bash"
+        run: |
+          export LANG="en_US.UTF-8"
+          sudo update-locale
+          sudo /etc/init.d/hpcc-init start
+
+      - name: Get test from Github
+        shell: "bash"
+        run: |
+          BUNDLES_TO_TEST=( "ML_Core" "PBblas" "GLM" "GNN" "DBSCAN" "LearningTrees" "TextVectors" "KMeans" "SupportVectorMachines" "LinearRegression" "LogisticRegression" )        
+
+          BUNDLES_COUNT=${#BUNDLES_TO_TEST[@]}
+          for ((i=0; i<$BUNDLES_COUNT; i++))
+          do
+              BUNDLE_NAME=${BUNDLES_TO_TEST[i]}
+              BUNDLE_REPO="https://github.com/hpcc-systems/${BUNDLES_TO_TEST[i]}.git"
+              INSTALL_CMD="ecl bundle install --update --force $ML_INSTALL_EXTRA ${BUNDLE_REPO}"
+          
+              tryCountMax=5
+              tryCount=$tryCountMax
+              tryDelay=1m
+
+              while true
+              do
+                  cRes=$( ${INSTALL_CMD} 2>&1 )
+                  retCode=$?
+                  if [[ $retCode -ne 0 ]]
+                  then
+                      tryCount=$(( $tryCount-1 ))
+
+                      if [[ $tryCount -ne 0 ]]
+                      then
+                          sleep ${tryDelay}
+                          continue
+                      else
+                          break;
+                      fi
+                  else
+                      BUNDLE_VERSION=$( echo "${cRes}" | egrep "^$BUNDLE_NAME" | awk '{ print $2 }' )
+                      break
+                  fi
+              done
+          done
+      - name: Run Tests
+        id: run
+        shell: "bash"
+        working-directory: /home/runner/.HPCCSystems/bundles/_versions/
+        run: |
+          ProcessLog()
+          { 
+              BUNDLE=$1
+              TARGET=$2
+              logfilename=$( ls -clr /home/runner/HPCCSystems-regression/log/thor.*.log | head -1 | awk '{ print $9 }' )
+              failed=$(cat ${logfilename} | sed -n "s/^[[:space:]]*Failure:[[:space:]]*\([0-9]*\)[[:space:]]*$/\1/p")                           
+          
+              if [[ "$failed" -ne 0 ]]
+              then 
+                  echo "Bundle : ${BUNDLE}" >> /home/runner/HPCCSystems-regression/log/Failed_test.summary
+                  cat ${logfilename} >> /home/runner/HPCCSystems-regression/log/Failed_test.summary
+                  echo "uploadArtifact=true" >> $GITHUB_OUTPUT
+              fi
+              # Rename result log file to name of the bundle
+              logname=$(basename $logfilename)
+              bundlelogfilename=${logname//thor/$BUNDLE}
+              printf "%s, %s\n" "$logname" "$bundlelogfilename"
+              mv -v $logfilename /home/runner/HPCCSystems-regression/log/ml-$bundlelogfilename
+          
+          }
+
+          BUNDLES_TO_TEST=( "ML_Core" "PBblas" "GLM" "DBSCAN" "GNN" "LearningTrees" "TextVectors" "KMeans" "SupportVectorMachines" "LinearRegression" "LogisticRegression" )        
```

Comment:
The 'BUNDLES_TO_TEST' was already defined, but yes in a different step. Define it once as a global and use anywhere where it needs (maintenance issue).